tests/unit/test_cli.py

```diff
@@ -189,6 +191,40 @@ def side_effect():
         mock_daemon_start.assert_called_once()
 
 
+    def test_ensure_directory_exists_not_exists(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            test_dir = pathlib.Path(tempdir).joinpath("test_ensure")
+            cli.ensure_directory_exists(str(test_dir))
```

Comment:
what is the reason for converting the `Path` object to a `str` here (and in other tests)?

Reply:
The type hints in the function signature ask for a `str` and the code that calls the function passes a `str`. I didn't test the `Path` objects, but I figured it would be better to stick with the type hints. I could add a comment explaining this, or if you don't think it's an issue (the tests do seem to pass without it), I can remove the conversion.

Reply:
I would remove the conversion. I expect that the type hints will be fixed in future versions of Python so that they take a `Path` object whenever a path is expected.

Reply:
Ok, I'll remove the conversion.